nixos/modules/system/boot/systemd/tmpfiles.nix

```diff
@@ -110,7 +112,7 @@ let
 
   # generates a single entry for a tmpfiles.d rule
   settingsEntryToRule = path: entry: ''
-    '${entry.type}' '${path}' '${entry.mode}' '${entry.user}' '${entry.group}' '${entry.age}' ${entry.argument}
+    '${entry.type}' '${path}' '${entry.mode}' '${entry.user}' '${entry.group}' '${entry.age}' ${escapeArgument entry.argument}
```

Comment:
Will this produce a breakage with already escaped values?

Reply:
It certainly will.

If we remove the `"\\"` element from the `escapeC` argument list, it should allow pre‐escaped arguments, but that feels a bit hacky.

grep’ing for typical escape sequences like `\x20` and `\x0a` as well as `escapeC` in nixpkgs shows that there are no pre‐escaped tmpfiles arguments used in nixpkgs itself.

Escaping the argument is rarely needed for most use cases and probably rarely used, so perhaps we could just check for `\\x[0-9A-Fa-f]{2}` occurrence in the option value and issue a warning.